core/src/Streamly/Internal/Data/Array/Generic.hs

```diff
@@ -146,7 +144,7 @@ fromPureStream x =
 fromCString# :: MonadIO m => Addr# -> m (Array Word8)
 fromCString# addr = fromStream $ D.fromCString# addr
 
-{-# DEPRECATED fromByteStr# "Please use fromCString instead" #-}
+{-# DEPRECATED fromByteStr# "Please use 'unsafeInlineIO . fromCString#' instead" #-}
```

Comment:
unsafePerformIO, unsafeInlineIO is an internal function.

Reply:
Then there should be a proper way to replicate the behaviour of `fromByteStr#`